solr/contrib/blob-directory/src/java/org/apache/solr/blob/BlobDirectory.java

```diff
@@ -181,7 +182,12 @@ protected void ensureOpen() throws AlreadyClosedException {
     @Override
     public void close() throws IOException {
       // Free the reference to the IndexOutput.
-      blobFileSupplier.indexOutput = null;
+      blobFileSupplier.getBlobFile();
+      // TODO
+      //  It could be possible to start pushing the file asynchronously with BlobPusher at this time,
```

Comment:
Nice comment but I wouldn't expect this idea to be in close(); I'd expect it to be somewhere when we write files.

Reply:
In this BlobIndexOutput we could intercept the writeXXX() methods to start pushing to the BlobPusher, this would require buffering. I'll add this option in the comment, and this will definitely require some perf benchmarking.